src/librustc/middle/check_match.rs

```diff
@@ -437,17 +437,15 @@ impl<'a, 'tcx> Folder for StaticInliner<'a, 'tcx> {
                 let def = self.tcx.def_map.borrow().get(&pat.id).cloned();
                 match def {
                     Some(DefConst(did)) => match lookup_const_by_id(self.tcx, did) {
-                        Some(const_expr) => {
-                            const_expr_to_pat(self.tcx, const_expr, pat.span).map(|new_pat| {
-
-                                if let Some(ref mut renaming_map) = self.renaming_map {
-                                    // Record any renamings we do here
-                                    record_renamings(const_expr, &pat, renaming_map);
-                                }
-
-                                new_pat
-                            })
-                        }
+                        Some(const_expr) => const_expr_to_pat(self.tcx,
```

Comment:
I think that this may wish to retain the outer block because it can be confusing if you expect the indented code below to be the body of the arm but it's in fact a closure.